ocaml-lsp-server/test/e2e/__tests__/textDocument-completion.test.ts

```diff
@@ -533,4 +533,177 @@ let x : t = \`I
       ]
     `);
   });
+
+  it("completion for holes", async () => {
+    await openDocument(outdent`
+let u : int = _
+`);
+
+    let items = await queryCompletion(Types.Position.create(0, 15));
+
+    expect(items).toMatchInlineSnapshot(`
+    Array [
+      Object {
+        "label": "0",
+        "textEdit": Object {
+          "newText": "0",
+          "range": Object {
+            "end": Object {
+              "character": 15,
+              "line": 0,
+            },
+            "start": Object {
+              "character": 14,
+              "line": 0,
+            },
+          },
+        },
+      },
+      Object {
```

Comment:
To make this test less noisy, how about we filter all completion candidates that start with `__`?